tests/trainer/test_trainer.py

```diff
@@ -2545,11 +2553,22 @@ def test_accelerator_config_from_yaml(self):
             self.assertEqual(trainer.accelerator.even_batches, False)
             self.assertEqual(trainer.accelerator.use_seedable_sampler, False)
 
+            # gradient accumulation kwargs wil configure the gradient_state
+            self.assertEqual(trainer.accelerator.gradient_state.plugin_kwargs["sync_each_batch"], True)
+
     def test_accelerator_config_from_dataclass(self):
         # Checks that accelerator kwargs can be passed through
         # and the accelerator is initialized respectively
+
+        grad_acc_kwargs = {
```

Comment:
Let's have a new test for `test_accelerate_config_gradient_accumulation` instead

Reply:
ok i called it `test_accelerate_config_from_dataclass_gradient_accumulation`